### PR TITLE
Viresh_Assignment_20_Done_(15+15+10+10)Mins

### PR DESCRIPTION
**Pull request by VireshJJun20**
@DeavinaSJun20 @KrishnaKTechnocredits : Please review and comment.

image

image

image

image

### CODE REVIEW

File: src/vireshJJun20/Assignment20/ArmstrongNumberCheck.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+		}
+		if(output==input)
+			System.out.println("The given number " + input + " is a armstrong number.");
+		else
+			System.out.println("The given number " + input + " is Not a armstrong number.");
+	}
+	
+	public static void main(String[] args) {
+		ArmstrongNumberCheck armstrongNumberCheck= new ArmstrongNumberCheck();
+		armstrongNumberCheck.getArmstrongNumber(153);
```
Comment (DeavinaSJun20) [inline src/vireshJJun20/Assignment20/ArmstrongNumberCheck.java]: Logic is perfect. On line 30 you can directly call the function like below,

new ArmstrongNumberCheck().getArmstrongNumber(153)`

File: src/vireshJJun20/Assignment20/FindDuplicateNumber.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+				} 
+			}
+		}
+		return -1;
+	}
+
+	public static void main(String[] args) {
+		FindDuplicateNumber findDuplicateNumber = new FindDuplicateNumber();
+		int[] ar = { 0, 1, 3, 2, 3 };
+		int DupNumber = findDuplicateNumber.getDuplicateNumber(ar);
```
Comment (DeavinaSJun20) [inline src/vireshJJun20/Assignment20/FindDuplicateNumber.java]: Logic looks good . Variable name should not start with Capital Letter on Line 29.

File: src/vireshJJun20/Assignment20/ReverseNumber.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+package vireshJJun20.Assignment20;
+
+public class ReverseNumber {
+
+	void reverseIntegers(int input) {
+		System.out.println("Input: " + input);
+		String output = "";
+		while (input > 0) {
+			int temp = input % 10;
+			output = output + String.valueOf(temp);
```
Comment (DeavinaSJun20) [inline src/vireshJJun20/Assignment20/ReverseNumber.java:16]: Here you are printing output as a string and not as an integer .The logic needs to be changed.

**Review by DeavinaSJun20 — CHANGES_REQUESTED** [review]
Make the changes as per suggestions

File: src/vireshJJun20/Assignment20/FindDuplicateNumber.java (hunk 2 above)
Comment (VireshJJun20) [reply src/vireshJJun20/Assignment20/FindDuplicateNumber.java]: I Agree!! updated..thanks.

image

File: src/vireshJJun20/Assignment20/ArmstrongNumberCheck.java (hunk 1 above)
Comment (VireshJJun20) [reply src/vireshJJun20/Assignment20/ArmstrongNumberCheck.java]: Yes, we can optimize in that  way. Thanks.
image

File: src/vireshJJun20/Assignment20/ReverseNumber.java (hunk 3 above)
Comment (VireshJJun20) [reply src/vireshJJun20/Assignment20/ReverseNumber.java:16]: I went with the requirements. Only mentioned about the input data type. but I had updated the output data type per your suggestion. Thanks.

image

File: src/vireshJJun20/Assignment20/ArmstrongNumberCheck.java (hunk 1 above)
Comment (DeavinaSJun20) [reply src/vireshJJun20/Assignment20/ArmstrongNumberCheck.java]: Code is working as expected.approved.

File: src/vireshJJun20/Assignment20/ReverseNumber.java (hunk 3 above)
Comment (DeavinaSJun20) [reply src/vireshJJun20/Assignment20/ReverseNumber.java:16]: Code is working as expected. 

Though you are getting correct output, we can improve this further as well by actually applying a maths logic which will help you not convert integer to string & vice versa

File: src/vireshJJun20/Assignment20/FindDuplicateNumber.java (hunk 2 above)
Comment (DeavinaSJun20) [reply src/vireshJJun20/Assignment20/FindDuplicateNumber.java]: Code is working as expected.approved.

**Review by DeavinaSJun20 — APPROVED** [review]

File: src/vireshJJun20/Assignment20/FindMissingNumber.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+			int missingNum = total - sum;
+			System.out.println("The missing number in between 1-10 is: " + missingNum);
+		} else {
+			System.out.println("The given array contains all 1-10 numbers.");
+		}
+	}
+
+	public static void main(String[] args) {
+		int[] input = { 1, 2, 3, 5, 6, 7, 8, 9, 10 };
+		int sumOfAll = 55;
```
Comment (KrishnaKTechnocredits) [inline src/vireshJJun20/Assignment20/FindMissingNumber.java:29]: Here you should write logic to find sum of first n number.
int al = array.length+1;
		int expectedSum = (al*(al+1))/2;

File: src/vireshJJun20/Assignment20/FindDuplicateNumber.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+package vireshJJun20.Assignment20;
+
+public class FindDuplicateNumber {
+	
+	int duplicateNum = 0;
+	int getDuplicateNumber(int[] input) {
+		int duplicateNum = 0;
+		for (int i = 0; i < input.length; i++) {
+			for (int j = i + 1; j < input.length; j++) {
+				if (input[i] == input[j]) {
```
Comment (KrishnaKTechnocredits) [inline src/vireshJJun20/Assignment20/FindDuplicateNumber.java:17]: Better to use a variable name as outerIndex and innerIndex.

File: src/vireshJJun20/Assignment20/ReverseNumber.java (hunk 3 above)
Comment (KrishnaKTechnocredits) [reply src/vireshJJun20/Assignment20/ReverseNumber.java:16]: You can refer this. Deavina is correct. 
void findReverseNumber(int number) {
		int reverseNum = 0;
		int mod =0;
		while(number>0) {
			mod = number%10;
			number = number/10;
		    reverseNum = mod + reverseNum*10 ;
		}
		System.out.println("Reverse Number is "+reverseNum);
	}

**Pull request merged**